src/tls.c

```diff
@@ -21,6 +21,10 @@
 #include <stdarg.h>
 #include <string.h>
 
+#if defined(__linux__)
+#include <sys/types.h>
+#endif
+
```

Comment:
```suggestion
#include <sys/types.h>
```

That `sys/types.h` defines `ssize_t` is specified in POSIX, so this should also be valid for BSD's and others ... I guess we could always include it here and only change it when we find a platform where it breaks.

Reply:
So some ifdef !window around it instead?